### PR TITLE
ESQL: Disable pushdown of WHERE past STATS

### LINKED ISSUE #115281: ESQL: ability to push down filter past `STATS` yields different result
Executing this query: `FROM employees | STATS v = VALUES(emp_no) by job_positions | WHERE job_positions == "Accountant" | MV_EXPAND v` returns:

```
       v       | job_positions 
---------------+---------------
10025          |Accountant     

```
In this case, the filter is pushed down to source, but through the `esql_single_query` query, so only SV `emp_no` matches (in this case the one doc with `emp_no` == 10025).

However, a semantically identical query: `FROM employees | STATS v = VALUES(emp_no) by job_positions | MV_EXPAND v | WHERE job_positions == "Accountant"` returns a different result:
```
       v       | job_positions 
---------------+---------------
10001          |Accountant     
10012          |Accountant     
10016          |Accountant     
10023          |Accountant     
10025          |Accountant     
10028          |Accountant     
10034          |Accountant     
10037          |Accountant     
10044          |Accountant     
10045          |Accountant     
10050          |Accountant     
10051          |Accountant     
10066          |Accountant     
10081          |Accountant     
10085          |Accountant     
10089          |Accountant     
10092          |Accountant     
10094          |Accountant     
```

The filter isn't pushed through `esql_single_query` and the agg will execute on SV and MV `emp_no`, so will yield different result.

### PR DESCRIPTION
**Pull request by alex-spies**
Fix https://github.com/elastic/elasticsearch/issues/115281

Let's disable the faulty optimization for now and re-introduce it later, correctly.

### CODE REVIEW

**Comment by alex-spies** [comment]
This still needs updating of the optimizer tests, but the fix to the production code is already in place.

**Review by luigidellaquila — COMMENTED** [review]
Let's see what the CI says, but the fix looks good 👍

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownAndCombineFilters.java [hunk 1]
```diff
@@ -38,18 +36,11 @@ protected LogicalPlan rule(Filter filter) {
         LogicalPlan child = filter.child();
         Expression condition = filter.condition();
 
+        // TODO: Push down past STATS if the filter is only on the groups; but take into account that `STATS ... BY field` performs an
+        // implicit `MV_EXPAND field`, which we cannot just push past.
```
Comment (luigidellaquila) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownAndCombineFilters.java:40]: Please add latest findings here, ie. that it performs and implicit MV_DEDUPE + MV_EXPAND

**Review by luigidellaquila — APPROVED** [review]
LGTM, thanks Alex!

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownAndCombineFilters.java (hunk 1 above)
Comment (alex-spies) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownAndCombineFilters.java]: Done!

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownAndCombineFilters.java [hunk 2]
```diff
@@ -38,18 +36,12 @@ protected LogicalPlan rule(Filter filter) {
         LogicalPlan child = filter.child();
         Expression condition = filter.condition();
 
+        // TODO: Push down past STATS if the filter is only on the groups; but take into account how `STATS ... BY field` handles
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownAndCombineFilters.java:39]: Should we add a link to https://github.com/elastic/elasticsearch/issues/115311?

**Review by bpintea — APPROVED** [review]
This breaks a previously working push of full text function filters. Something like 
`FROM employees | STATS v = VALUES(emp_no) by job_positions | WHERE MATCH(job_positions, "Accountant")`
fails now in a similar way to https://github.com/elastic/elasticsearch/issues/115353.
We have no tests for it working, so not visible in this PR.

But I guess this should go in and that addressed separately.

**Pull request merged**